java/src/main/java/com/google/appengine/tools/mapreduce/impl/pipeline/DeleteFilesJob.java

```diff
@@ -1,40 +1,47 @@
 package com.google.appengine.tools.mapreduce.impl.pipeline;
 
-import static com.google.appengine.tools.mapreduce.impl.MapReduceConstants.GCS_RETRY_PARAMETERS;
 
-import com.google.appengine.tools.cloudstorage.GcsService;
-import com.google.appengine.tools.cloudstorage.GcsServiceFactory;
-import com.google.appengine.tools.cloudstorage.RetriesExhaustedException;
+import com.google.appengine.tools.mapreduce.GcpCredentialOptions;
 import com.google.appengine.tools.mapreduce.GcsFilename;
 import com.google.appengine.tools.pipeline.Job1;
 import com.google.appengine.tools.pipeline.Value;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import lombok.RequiredArgsConstructor;
 
-import java.io.IOException;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
  * A job which deletes all the files in the provided GoogleCloudStorageFileSet
  */
+@RequiredArgsConstructor
 public class DeleteFilesJob extends Job1<Void, List<GcsFilename>> {
 
   private static final long serialVersionUID = 4821135390816992131L;
-  private static final GcsService gcs = GcsServiceFactory.createGcsService(GCS_RETRY_PARAMETERS);
   private static final Logger log = Logger.getLogger(DeleteFilesJob.class.getName());
 
+  private final GcpCredentialOptions credentialOptions;
+
+
   /**
    * Deletes the files in the provided GoogleCloudStorageFileSet
    */
   @Override
   public Value<Void> run(List<GcsFilename> files) throws Exception {
-    for (GcsFilename file : files) {
-      try {
-        gcs.delete(new com.google.appengine.tools.cloudstorage.GcsFilename(file.getBucketName(), file.getObjectName()));
-      } catch (RetriesExhaustedException | IOException e) {
-        log.log(Level.WARNING, "Failed to cleanup file: " + file, e);
+    Storage storage = GcpCredentialOptions.getStorageClient(credentialOptions);
+    try {
+      List<Boolean> results = storage.delete((BlobId[]) files.stream().map(GcsFilename::asBlobId).toArray());
+      for (int i = 0; i < results.size(); i++)  {
+        if (!results.get(i)) {
+          log.log(Level.WARNING, "Failed to cleanup file: " + files.get(i));
```

Comment:
I'd add as a comment that in case of  `false` return for the specified blob, that could means [that](https://googleapis.dev/java/google-cloud-storage/latest/index.html) *If a blob was not found, deletion failed or access to the resource was denied the corresponding item is false.*